packages/multistore/src/extension.ts

```diff
@@ -1,39 +1,40 @@
-import { Request } from "express";
-import { CacheManager, ExtensionParams } from "./types";
+import { ApiClientExtension } from "@vue-storefront/middleware";
+import { MultistoreExtensionMethods } from "./types";
 import { resolveDomain } from "./resolve/resolveDomain";
 import { validateMultistoreMethods } from "./validate/validateMultistoreMethods";
 import { requiredMethodsErrors } from "./validate/requiredMethodsErrors";
 import { fetchConfigWithCache } from "./cache/fetchConfigWithCache";
 
-let cacheManager: CacheManager;
+export const createMultistoreExtension = (
+  multistoreConfig: MultistoreExtensionMethods
+) => {
+  const cacheManager = multistoreConfig.cacheManagerFactory();
 
-export const multistoreExtension = {
-  name: "multistore-extension",
+  return {
+    name: "multistore-extension",
+    isNamespaced: false,
+    extendApp: () => {
+      Object.keys(requiredMethodsErrors).forEach((requiredMethod) => {
+        validateMultistoreMethods(requiredMethod, multistoreConfig);
+      });
```

Comment:
It was like that before, but I learned funny thing on that:
<img width="771" alt="image" src="https://github.com/vuestorefront/vue-storefront/assets/35867383/75c345a2-f410-4401-91e9-6e51b3d65532">
Here's an explanation by copilot:
> You're correct. In JavaScript and TypeScript, a for...in loop will iterate over all enumerable properties of an object, including those in the prototype chain. This can sometimes lead to unexpected behavior if the object's prototype has enumerable properties. To avoid this, you can add an if statement inside the loop to check if the property belongs to the object itself and not to its prototype. You can do this using the Object.prototype.hasOwnProperty() method.
But the version with if was ugly so I went with forEach xd